internal/api/handler/create_stage_v1alpha1.go

```diff
@@ -49,14 +50,23 @@ func CreateStageV1Alpha1(
 		default:
 			return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("environment should not be empty"))
 		}
-		if err := kc.Create(ctx, &stage); err != nil {
+
+		var ns corev1.Namespace
+		if err := kc.Get(ctx, client.ObjectKey{Name: stage.GetNamespace()}, &ns); err != nil {
 			if kubeerr.IsNotFound(err) {
 				return nil, connect.NewError(connect.CodeNotFound, errors.Errorf("project %q not found", stage.GetNamespace()))
 			}
+			return nil, connect.NewError(connect.CodeInternal, errors.Wrap(err, "failed to get project"))
+		}
+		if ns.GetLabels()["kargo.akuity.io/project"] != "true" {
+			return nil, connect.NewError(connect.CodeFailedPrecondition, errors.Errorf("namespace %q is not a project", stage.GetNamespace()))
+		}
```

Comment:
not an immediate need but a future improvement would be to just maintain a shared informer for these project namespaces. I see we need the namespaces very frequently in the server so might just be better to cache them instead of querying for them in every place.

Reply:
@gdsoumya he's using the client from controller runtime and that has a cache and shared informer already built in.